### ISSUE TITLE
ECONNREFUSED not thrown correctly

### ISSUE THREAD

**Issue by joffray589** [typeorm/typeorm#610]
I created a sample repository (https://github.com/joffray589/typeorm-sample) which expose this problem, here is the main.js
````
let dbOptions: ConnectionOptions = { /* ... */ };

(async () => {
    try{
        let connection = await createConnection(dbOptions); // (1)
        console.log("connection.isConnected = " + connection.isConnected);
        await connection.getRepository(FooEntity).find();       // (2)
        console.log("Mysql connection ok");
    }
    catch(e){
        console.error("Mysql connection error : " + e );
    }
})();
````

If I stop mysql server and run this example with autoSchemaSync=true, ECONNREFUSED will be thrown at (1).
But if i set autoSchemaSync to false and run this example again, ECONNREFUSED will be thrown at (2) and connection.isConnected will return true.

Here is my environnement: 

Windows 7,
typeorm: 0.0.10,
mysql: 2.13.0

**Comment by joffray589** [comment]
After some investigations, I think that this issue comes from the MysqlDriver.connect() method. 
The pool attribute is instanciated but the connection should be tested after.
I think this issue should be resolved by adding this code at the end of the connect method :
 ```
        return new Promise<void>((resolve, reject) => {
            this.pool.getConnection(function(err, connection) {
                if(err){
                    reject("Mysql connection error : " + err);
                }
                else{
                    resolve();
                }
            });
        });
```

I'm going to make some tests and I'll make a pull request if it works.

**Comment by joffray589** [comment]
And here is the pull request : #618

**Comment by pleerock** [comment]
fixed by #619
